protocol.c

```diff
@@ -62,6 +62,9 @@ dsmesock_connection_t* dsmesock_connect(void)
 
       memset(&c_addr, 0, sizeof(c_addr));
       c_addr.sun_family = AF_UNIX;
+      if (strlen(dsmesock_filename) >= sizeof(c_addr.sun_path)) {
+          return NULL;
```

Comment:
Simply doing a mid-function return -> leaks socket fd.

I'd suggest moving the test and return earlier -> as soon as dsmesock_filename is known, but no dynamic resources have been allocated yet.